### DISCUSSION TITLE
Feature Request for added dashboard api creation functionality

### DISCUSSION THREAD

**Discussion by nogreintel** [apache/superset#18543] [Ideas]
**Is your feature request related to a problem? Please describe.**
When using the superset api to create dashboards, the current api only accepts the following inputs:
`{
  "css": "string",
  "dashboard_title": "string",
  "json_metadata": "string",
  "owners": [
    0
  ],
  "position_json": "string",
  "published": true,
  "roles": [
    0
  ],
  "slug": "string"
}`
If one wants to create a dashboard entirely defined in this single api call for a post to /dashboards/, even when defining the relevant charts in position_json, those charts are not recognized, and show up as non existent in superset, because the api doesn't accept the necessary data for slices and datasources.

**Describe the solution you'd like**
It would be ideal if the superset api for post to /dashboards/ would accept the following parameters:
`{
  "css": "string",
  "dashboard_title": "string",
  "json_metadata": "string",
  "owners": [
    0
  ],
  "position_json": "string",
  "published": true,
  "roles": [
    0
  ],
  "slug": "string",
  "slices": []
  "datasources": []
}`

**Describe alternatives you've considered**
The current workaround is to define a superset dashboard in a json template and then use the /dashboard/import api. 

**Additional context**

**Comment by geido** [comment]
Hello @nogreintel thanks for the suggestion!

I see the point that you are making and I understand it. However, I believe the endpoint is fine as it is to avoid the risk of polluting the endpoint with a lot of data. I am curious to learn what some other people think. @villebro what's your take?

**Comment by nogreintel** [comment]
I understand your perspective. The thing I'd like to emphasize is that this prevents any clean way of automating dashboard creation. 

Yet the equivalent functionality can be had by posting with a dashboard import (using a json dashboard_template)--and even this poses some development issues (like having to parse the file and also ensuring new chart IDs are generated properly). So effectively the end point can still be polluted with the same amount of data, so we may as well do it in a more user friendly way. 

Please let me know your thoughts.

**Reply by liberty-rising** [reply]
Were you able to at least implement the json alternative?

**Comment by liberty-rising** [comment]
This should definitely be added. Dashboard automation works up to the dashboard creation portion. You can automate the connection to the database, creating the data source, and creating the charts. It only makes sense with the current workflow to also allow support for automating dashboard creation @geido @villebro
